imap/dav_db.c: undeclared identifier 'rrock' when building with WITH_DAV but without WITH_JMAP

## Summary

`imap/dav_db.c` fails to compile when JMAP is disabled at configure time, because the `rrock` variable is referenced from code blocks that are not gated by `WITH_JMAP`.

## Affected version

- 3.13.4 (cyrus-imapd-3.13 branch)
- Looks like it was introduced when the `WITH_JMAP` `#ifdef` was tightened around the `rrock` declaration in `_dav_reconstruct_mb()`.

## Build configuration that triggers the bug

Anything that ends up with `WITH_JMAP` undefined but with `WITH_DAV` (or `USE_SIEVE`) defined.

`./configure --disable-jmap` (and HTTP off, since HTTP requires JMAP) is enough to reproduce.

## Error

```
imap/dav_db.c:431:39: error: use of undeclared identifier 'rrock'
  431 |         if (!r) r = addproc(mailbox, &rrock->cmodseqs);
      |                                       ^
1 error generated.
gmake[2]: *** [Makefile:6043: imap/libcyrus_imap_la-dav_db.lo] Error 1
```

## Root cause

In `imap/dav_db.c`, the declaration of `rrock` is gated on `WITH_JMAP` only:

```c
static int _dav_reconstruct_mb(const mbentry_t *mbentry,
                               void *rock
#ifndef WITH_JMAP
                                          __attribute__((unused))
#endif
                              )
{
#ifdef WITH_JMAP
    struct recon_rock *rrock = (struct recon_rock *) rock;
    struct buf attrib = BUF_INITIALIZER;
#endif
    ...
}
```

But `rrock` is also used in:

- the `#ifdef WITH_DAV` block (`rrock->last_mbtype`, `rrock->cmodseqs`), and
- unconditionally at line 431: `r = addproc(mailbox, &rrock->cmodseqs);`

The `struct recon_rock` itself is defined unconditionally, so the cast is always valid — the `#ifdef` on the declaration is the only thing that breaks the build.

## Suggested patch

```diff
--- a/imap/dav_db.c
+++ b/imap/dav_db.c
@@ -365,16 +365,16 @@ struct recon_rock {
  * mboxlist_usermboxtree() callback function to create DAV DB entries for a mailbox
  */
 static int _dav_reconstruct_mb(const mbentry_t *mbentry,
                                void *rock
-#ifndef WITH_JMAP
+#if !defined(WITH_DAV) && !defined(WITH_JMAP) && !defined(USE_SIEVE)
                                           __attribute__((unused))
 #endif
                               )
 {
-#ifdef WITH_JMAP
     struct recon_rock *rrock = (struct recon_rock *) rock;
+#ifdef WITH_JMAP
     struct buf attrib = BUF_INITIALIZER;
 #endif
     int (*addproc)(struct mailbox *, hashu64_table *) = NULL;
```

This declares `rrock` whenever any backend (`WITH_DAV`, `WITH_JMAP`, `USE_SIEVE`) is enabled, and only marks `rock` as `unused` when none of them are. `struct buf attrib` remains JMAP-only since it is only used inside the `WITH_JMAP` block.

## Reproduction

```sh
./configure --disable-http --disable-jmap --enable-replication \
            --enable-sieve --with-openssl
gmake
```

--- a/imap/dav_db.c
+++ b/imap/dav_db.c
@@ -396,15 +396,12 @@ struct recon_rock {
  */
 static int _dav_reconstruct_mb(const mbentry_t *mbentry,
                                void *rock
-#ifndef WITH_JMAP
+#if !defined(WITH_DAV) && !defined(WITH_JMAP) && !defined(USE_SIEVE)
                                           __attribute__((unused))
 #endif
                               )
 {
-#ifdef WITH_JMAP
     struct recon_rock *rrock = (struct recon_rock *) rock;
-    struct buf attrib = BUF_INITIALIZER;
-#endif
     int (*addproc)(struct mailbox *, hashu64_table *) = NULL;
     int writelock = 0;
     int r = 0;
@@ -435,7 +432,8 @@ static int _dav_reconstruct_mb(const mbentry_t *mbentry,
         addproc = &mailbox_add_email_alarms;
         break;
 
-    case MBTYPE_EMAIL:
+    case MBTYPE_EMAIL: {
+        struct buf attrib = BUF_INITIALIZER;
         r = annotatemore_lookup(mbentry->name, "/specialuse", rrock->userid, &attrib);
         if (!r && buf_len(&attrib)) {
             strarray_t *specialuse =
@@ -448,6 +446,7 @@ static int _dav_reconstruct_mb(const mbentry_t *mbentry,
         }
         buf_free(&attrib);
         break;
+    }
 #endif
     }
 

--- a/imap/ical_support.h
+++ b/imap/ical_support.h
@@ -251,8 +251,6 @@ extern void icalcomponent_set_jmapid(icalcomponent *comp, const char *id);
 
 extern bool ical_is_valid_color(const char *val, bool allow_alpha);
 
-#endif /* HAVE_ICAL */
-
 /**
  * Looks up a property parameter by name.
  *
@@ -266,4 +264,7 @@ extern bool ical_is_valid_color(const char *val, bool allow_alpha);
  */
 extern vcardparameter *vcardproperty_get_parameter_by_name(vcardproperty *prop,
                                                            const char *name);
+
+#endif /* HAVE_ICAL */
+
 #endif /* ICAL_SUPPORT_H */

